crates/pallet-subspace/src/lib.rs

```diff
@@ -494,8 +491,9 @@ mod pallet {
 
         /// Submit new segment header to the blockchain. This is an inherent extrinsic and part of
         /// the Subspace consensus logic.
+        /// TODO: constraint the number of `SegmentHeader` within a single extrinsic
```

Comment:
Do you mean using `BoundedVec<SegmentHeader>` in specific?

Reply:
Using `BoundedVec<SegmentHeader>` as an extrinsic parameter may not help as it is constructed by the user, I mean having something like `Config::MaxNumberSegmentHeader` and do the checking in `validate_unsigned/pre_dispatch`.

Reply:
Yeah, that's what I meant, not the argument in the call, using BoundedVec for the inner `do_store_segment_headers`, then the extrinsic simply fails when it fails to be converted to BoundedVec, which is typically constrained by a parameter in Config. I don't think it's a bad idea, but Nazar may have more thoughts.

Reply:
The actual limit is 1 with current parameters, but it might change in the future. 1024 is impossible as far into the foreseeable roadmap as I can comment on.

Reply:
Should we make this limit explicit (i.e. add `Config::MaxSegmentHeader` and some check)? Or I can also just use something like 128 to re-benchmark this extrinsic and update the result.

Reply:
I don't think we need to make it explicit, benching to something like 20 should be sufficient to get an idea of added cost of additional segment header.